### PR TITLE
📢 Mentee Public Profile

### LINKED ISSUE #290: Mentee public profile page
#### Description
image

### PR DESCRIPTION
**Pull request by Leundai**
## Description :star2:

Fixes #290 
- Generalized public profile to support mentee profile
- updated app.js with redirect
- connected to backend

## Screenshots:
image

### CODE REVIEW

File: frontend/src/components/ProfileContent.js [hunk 1]
```diff
@@ -1,127 +1,140 @@
...
+};
+
+const getSpecializationTags = (specializations) => {
+  return specializations.map((specialization, idx) => (
+    <div className="mentor-specialization-tag">{specialization}</div>
+  ));
+};
+
+const getEducations = (educations) => {
+  if (!educations || !educations[0]) {
```
Comment (kenetec) [inline frontend/src/components/ProfileContent.js:37]: You should check for `educations.length` before checking for `educations[0]`. You could also set a default value to educations like `const getEducations = (educations = []) => {`

File: frontend/src/components/ProfileContent.js [hunk 2]
```diff
@@ -1,127 +1,140 @@
...
-    return languages.join(" • ");
-  };
-
-  const getSpecializationTags = (specializations) => {
-    return specializations.map((specialization, idx) => (
-      <div className="mentor-specialization-tag">{specialization}</div>
-    ));
-  };
+  const getProfileButton = () => {
+    if (props.isMentor)
```
Comment (kenetec) [inline frontend/src/components/ProfileContent.js:65]: This can also be grabbed from `useAuth()` which might be cleaner to do.

**Review by kenetec — APPROVED** [review]
siiick 😎  the CSS just needs some fixing on the mentee profile page but otherwise it looks amazing. Also, I hijacked your PR to include some routing fixes for getting to `/mentee-appointments` and to fix a loading issue with the Gallery page for mentors.

**Review by AngelaLuo49021 — APPROVED** [review]
Cute!

**Pull request merged**